### DISCUSSION TITLE
All devices show in Homebridge, but most don't work in HomeKit

### DISCUSSION THREAD

**Discussion by JeffGoldner** [iklein99/homebridge-smartthings#103] [General]
Probably premature as to what is supported. All my devices were discovered but most don't work in HK.

Here's what I have:

Schlage Locks - one using the standard Edge driver, one use Rboy DTH. Rboy works, Edge "Not Supported"
Ecolink Motion Sensors - using Edge driver. "Not supported" by HomeKit
Inovelli Blue (Zigbee) - using their beta Edge driver. "Not supported" by HomeKit
GE Jasco Dimmers - some use Edge driver, others use legacy DTH. "Not supported"
GE Jasco On/Off switches - some use Edge driver, some legacy DTH.  Most "Not supported"
TVs - "Not supported" - don't care really.

Is it too early to be using this plug-in? I'd like to get away from the v2 one since I don't know when ST will kill it. I'd rather not switch hubs, if that's even possible, to get HK support.

Thanks

**Comment by iklein99** [comment]
The plugin supports edge.

The switches, locks, motion sensors, dimmers should work without an issue.

TV isn’t supported.

Nots sure what the Invelli Blue is.

**Reply by JeffGoldner** [reply]
This is what shows up in Home and none of the "Not Supported" devices can be managed (turn on/off, dimmed, etc.)

https://1drv.ms/u/s!AsO_d59wuLnNnd8DKOMfSqlYqivRGw?e=ujerti

Inovelli Blue is a Zigbee switch.

**Comment by iklein99** [comment]
Can’t tell anything from looking at the device screen.

Assume you have successfully installed and set up the plugin.  Do the following and send me the log file.  Don’t post it to GitHub.

Clear your logs (truncate).  Set Debug mode in the Homebridge setting menu and restart.

After about 30 seconds, download the logs and send to me at ***@***.*** ***@***.***>.

Also list the names of these devices so I can find them in the log.

You should then reset debug mode and restart homebridge.

**Reply by JeffGoldner** [reply]
I restarted the plug-in as a child bridge, and acquired yet another token with all permissions, although I originally had what should be sufficient - all device perms were selected before. If you have specific requirements, please post them.

Now I can add all the devices after adding the bridge to Home, and control them. But behavior isn't great. Some devices that were on shut off by themselves, and everything is really slow compared to the original plug-in (which was disabled during this test) and often devices go to Not Responding. For now, I am going to stay with the other plug-in where everything has been working  for several years. If/when SmartThings cuts me off, I might try again.

Thanks

**Reply by steinauf85** [reply]
that v2 plugin is based on groovy and will be killed after 12/31. 

this plugin is based on edge and will remain alive going forward, but is not as mature as the v2 with the all of the capabilities and supported devices.

**Comment by coryabooth** [comment]
I have similar devices.  Agreed the locks were a pain, but I got all devices working.  One big key, are you using SmartThings CLI (command-line) to inspect and modify drivers?  Using "SmartThings edge:drivers:switch" you can flip flop drivers and test.

**Comment by iklein99** [comment]
I am working on a version that leverages web hooks for updates which will address the slow update and eliminate the need to constantly poll devices.  However, that will not be free.  I will be running a service in the cloud which I have to pay for.  I expect it to be a nominal fee.  I will not be forcing people to use it if they are ok with the current functionality.

**Comment by kevininor** [comment]
I'm seeing the same issue reported in bug https://github.com/iklein99/homebridge-smartthings/issues/123#issue-1536594910.
